### DISCUSSION TITLE
Disable predefined keymaps

### DISCUSSION THREAD

**Discussion by badwatt** [olimorris/codecompanion.nvim#240] [Ideas]
Hi @olimorris, thank you very much for such an amazing piece of software!

I've been having a hard time finding out how to disable predefined keymaps, as they are conflicting with my previously defined keymaps.

I think it would be awesome to have an option to disable them.

Once again, thank you!

**Comment by olimorris** [comment]
Hey @alvaro17f, thanks for your kind words and hope you're enjoying CodeCompanion.

The problem with disabling the keymaps is you will lose all ability to use the plugin. Unfortunately, it's not as simple as you being able to call something like `:require("codecompanion.chat").submit()`. The keymaps are only scoped to chat buffers and/or buffers that you have initiated the Inline Assistant from. So there shouldn't be too much of clash. Even if there is, you can update the config to select another keymap relatively easily.
